timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockWatchingService.java

```diff
@@ -27,7 +28,8 @@
 public interface LockWatchingService {
     void startWatching(LockWatchRequest locksToWatch);
     LockWatchStateUpdate getWatchStateUpdate(OptionalLong lastKnownVersion);
-
+    LockWatchStateUpdate getWatchStateUpdate(OptionalLong lastKnownVersion, long endVersion);
+    <T> ValueAndVersion<T> runTaskAndAtomicallyReturnVersion(Supplier<T> task);
```

Comment:
maybe verbose, but would prefer `...WatchStateVersion` perhaps given that this is on the super-interface  `AsyncTimelockService`